adafruit_cursorcontrol/cursorcontrol_cursormanager.py

```diff
@@ -118,24 +126,34 @@ def _read_joystick_y(self, samples=3):
         reading = 0
         # pylint: disable=unused-variable
         if hasattr(board, "JOYSTICK_Y"):
-            for sample in range(0, samples):
+            for _ in range(0, samples):
                 reading += self._joystick_y.value
             reading /= samples
         return reading
 
-    def _check_cursor_movement(self, pressed=None):
-        """Checks the PyBadge D-Pad or the PyGamer's Joystick for movement.
-        :param int pressed: 8-bit number with bits that correspond to buttons
-            which have been pressed down since the last call to get_pressed().
+    def _store_button_states(self, event):
+        """Stores the state of the PyBadge's D-Pad or the PyGamer's Joystick
+        into a byte
+
+        :param Event event: The latest button press transition event detected.
         """
+        if event:
+            bit_index = event.key_number
+            current_state = self._pad_states >> bit_index
```

Comment:
I think this misses a mask `& 0x01`, otherwise the states above the one you are interested in are going to interfere

Reply:
Yes, thank you!